src/model.rs

```diff
@@ -422,7 +430,18 @@ impl Model<ProblemCreated> {
 
 impl Model<Solved> {
     fn _set_best_sol(&mut self) {
-        if self.scip.get_status() == Status::Optimal {
+        // a best solution can exist even if the status is not optimal, for example if the 
+        // time limit is reached. In this case, the best solution is the best solution explored,
+        // assuming any viable solutions had been explored.
+
+        // to check if any viable solutions had been explored and to fetch the best solution,
+        // the problem needs to be in a stage where it has variables and constraints.
+        if !self.scip.has_vars_and_constraints() {
+            return;
+        }
+
+        // if at least one viable solution had been explored, get the best such solution.
```

Comment:
Unecessary comment, your code is perfectly understandable without :)

Reply:
removed